smartcosmos-framework/src/main/java/net/smartcosmos/SmartCosmosErrorState.java

```diff
@@ -0,0 +1,46 @@
+package net.smartcosmos;
+
+/**
+ * An enum of the error states that may be returned from SmartCosmos services.
+ */
+public enum SmartCosmosErrorState {
+    SUCCESS(1, "Success"),
+    FAILURE(0, "Failure"),
+    NOT_AUTHORIZED(-1, "Authenticated user is not authorized to make this call"),
+    NOT_FOUND(-2, "No such URN"),
+    NOT_LICENSED(-3, "Unlicensed feature"),
+    UNKNOWN_ENTITY_TYPE(-4, "Unknown Entity Reference Type"),
+    MISSING_REQUIRED_FIELD(-5, "JSON is missing a required field: <…>"),
```

Comment:
Previously `-5` was also used for constraint violations, which includes missing required fields. The message therefore should be `JSON is missing required fields or violates constraints: <...>`